Add: page on github issues in our organization (#67)
pyOpenSci/handbook: community/github/issues.md

# pyOpenSci GitHub issue process

> When possible, pull request and issue submissions and reviews should follow

Reviewer: ```suggestion
When possible, pull requests, issue submissions and reviews should follow
```

standard open source workflows. Below are guidelines for handling issues.

## Guidelines for New Issues

> **Issues in should be as specific as possible:** specifics within an issue help both our future selves and also outside contributors understand the goal or

Reviewer: ```suggestion
**Issues should be as specific as possible:** specifics within an issue help both our future selves and also outside contributors understand the goal or
```

desired outcome associated with addressing the issue. This is important both internally and for issues that we tag as `help-wanted`, which we hope community members will address in pyOpenSci sprints.

To ensure an issue is well-written and specific, include the following details:

- **Clear Title:** A concise and descriptive title summarizing the issue or feature request helps us scan through issues and understand what each issue is about. Below are some example titles that are specific

>     * `Bug: broken link in link-to-page-here / name of page / document etc`

Reviewer: LOVE that there are examples!

    * `Fix: confusing paragraph on Python packaging with Hatch`
    * `Add: page on using pixi for dependencies`

- **Description:** A detailed explanation of the issue or feature request, including context, background information, and the reason for the request. Explain why the issue is important and what problem it solves.

> - **Screenshots/Code Samples:** Screenshots, code snippets, links, or any other relevant files that can help in understanding the issue better.

Reviewer: ```suggestion
- **Screenshots/Code Samples:** Include screenshots, code snippets, links, or any other relevant files that can help others in understanding the issue better.
```

> - **Possible Solutions/Recommendations:** Any ideas or suggestions for how to address the issue, which can help guide contributors towards a solution.

Reviewer: ```suggestion
- **Possible Solutions/Recommendations:** It's helpful to provide any ideas or suggestions for how to address the issue, which can help guide contributors towards a solution.
```

> - **Related Issues or Pull Requests:** References to any related issues or pull requests, providing additional context and understanding of the broader scope of the issue.

Reviewer: nit: issues and pull requests are capitalized here, but earlier in the post they're lowercase. I believe that we generally keep them lowercase throughout our posts, unless we type `PR`.

> - **Related Issues or Pull Requests:** References to any related issues or pull requests, providing additional context and understanding of the broader scope of the issue.

Reviewer: ```suggestion
- **Related Issues or Pull Requests:** Add references to any related issues or pull requests, which provides additional context and understanding of the broader scope of the issue.
```

### If you are reporting a code bug

> * **Steps to Reproduce (for bugs):** A step-by-step guide on how to reproduce the issue, including relevant code snippets, commands, or configurations.

Reviewer: ```suggestion
* **Steps to Reproduce (for bugs):** A step-by-step guide (generally a list or narrative) on how to reproduce the issue, including relevant code snippets, commands, or configurations.
```

> * **Expected vs. Actual Behavior (for bugs):** A description of what you expected to happen and what actually happened. This helps in understanding the discrepancy and the impact of the bug.

Reviewer: nit: we're flipping between title case and sentence case throughout the post, and should choose just one (generally we go with sentence case).

> * **Environment Details (for bugs):** Details about the environment where the issue was observed, such as operating system, Python version, library versions, and any other relevant software/hardware details.

Reviewer: would it be helpful to tell people where to get some of this information? I'm thinking of beginners who might be using a tutorial, and not necessarily know how to get their Python or library versions, and/or know what additional software/hardware details might be helpful.

Author: normally you would do that in an issue template - where you'd have -- provide conda information for instance and then you have a code example. It will be different for each repo depending on what the repo is - ie sphinx book vs jekyll website ruby vs python. 

the ideal appoach would be to use issue forms for this.

### If you have permissions, label the issue

> While outside contributors will not have permission to label issues, pyOpenSci core team members and volunteers will. Be sure to add appropriate labels to

Reviewer: it would be helpful to have a quick list here on what our labels are and what they're used for

Author: we have an open help-wanted ticket on the labels. the labels are different in each repo. 

https://github.com/orgs/pyOpenSci/projects/3?pane=issue&itemId=62313858

so while this is a good idea it's a lot more complicated than a quick list! we will need to complete the task abbove which involved deciding what labels we want to curate and syncing them across repos! jon did a bunch of work on this at our scipy sprint. you can open a new issue about this here that will be a future issue when we get that workflow setup  if you wish.

issues to make it easier to triage them.

## Help-wanted / sprintable issues

If an issue is something that anyone in the community could potentially
address, it's ideal to label the issue with `help-wanted` and/or `sprintable`.
A sprintable issue refers to an issue that could be completed or worked on
during a 1-2 day sprint, thus it should be smaller and more confined in scope.
A help-wanted issue is one that anyone is welcome to work on during whatever
time they have available.

Once the `help-wanted` or `sprintable` issue label is added, the issue will be
automatically added to our
[pyOpenSci help-wanted board](https://github.com/orgs/pyOpenSci/projects/3).

> This automation is implemented currently for a single repository (the packaging guide) but we plan to implement it for other repositories using the add-to-project GitHub action.

Reviewer: ```suggestion
This automation is implemented currently for a single repository (the packaging guide), but we plan to implement it for other repositories using the add-to-project GitHub action.
```

:::{note}
The issue will be archived from the project board once it is closed.
:::

:::{todo}
Add link to sprints page when it's online
:::

If an issue is unclear, a pyOpenSci staff member or designated community
member can ask the issue author to provide more information.

:::{todo}
Add section on labels
:::

## Issue Maintenance

Our goal at pyOpenSci is to keep our list of issues current and active.

> Quarterly issue cleanup sessions should be implemented to ensure issues are

Reviewer: ```suggestion
Quarterly issue cleanup sessions are implemented to ensure issues are
```

> either being actively addressed or to assess whether older or stale issues can

Reviewer: ```suggestion
either being actively addressed, or to assess whether older or stale issues can
```

> be potentially closed.

Reviewer: ```suggestion
potentially be closed.
```
